symengine/basic.h

```diff
@@ -278,5 +287,17 @@ struct hash<SymEngine::Basic>;
         return type_code_id;                                                   \
     };                                                                         \
     SYMENGINE_INCLUDE_METHODS(;)
+#else
+#define IMPLEMENT_TYPEID(ID)                                                   \
+    /*! Type_code_id shared by all instances */                                \
+    const static TypeID type_code_id = ID;                                     \
+    SYMENGINE_INCLUDE_METHODS(;)
+#endif
+
+#ifdef USE_VIRTUAL_TYPEID
+#define ASSIGN_TYPEID()
```

Comment:
Change this to `SYMENGINE_ASSIGN_TYPEID`. It's always better to have `SYMENGINE` in macros.